block-layer/graphsync/graphsync.md

```diff
@@ -59,6 +59,99 @@ type GraphSyncNet interface {
 
 ## Network Messages
 
+### Protocol Version 1.1.0
+
+Graphsync network messages are encoded in DAG-CBOR. They have the following schema
+
+```ipldsch
+type GraphSyncExtensions {string:Any}
```

Comment:
`string` should be `String` in this position. I think `Any` should be fine for now if it can't be enumerated or properly pinned down, `Any` just hasn't fully filtered through our specs or stack because it's quite tricky. The original pb has `bytes` so might that be a safe choice for now? It could be turned into a union later or enumerated in some other way if needed. What types of things go into this?

Reply:
sorry, just re-read your comment about why you didn't use bytes! ignore that bit then, but I'm still interested in what kind of data this might contain and if it could be enumerated? could it be limited to scalar values to rule out maps and lists perhaps? we have `AnyScalar` for that case if it works.

Reply:
Confirm `Any` should be fine.

(We still have some things to wade through in https://github.com/ipld/specs/issues/318, it seems, but I think it's something we'll have to reach a conclusion on sooner or later, and I don't think there's any way we won't have an `Any` by the end.)

Reply:
It definitely isn't AnyScalar. We already use arrays and maps.

There are some other challenges for actually writing this in code-- the string represents the extension name, and the type is based on that, but we support user defined extensions, so it's not possible to know everything ahead of time.

There is a whole concept in `cbor-gen` world of `Deferred` -- where you don't deserialize till later, leaving in byte representation till you know what you want to deserialize to. I'm not sure this is the right approach for IPLD. But this will probably come up as well with https://github.com/ipld/go-ipld-adl-hamt -- I see there you've defined an Any type, but many HAMTs have a specific type at the bottom -- people may want to paramaterize their HAMTs and use code gen'd nodes for the day in the leaves. We should probably figure out how this might work.

Anyway though, just putting Any here at least allows me to define the schema